win/WIN_ColourSlider.cpp

```diff
@@ -0,0 +1,50 @@
+#include "WIN_ColourSlider.h"
+
+using namespace win;
+
+ColourSlider::ColourSlider(SDL_Renderer* renderer, gfx::Rectangle rect, const char* name, gfx::Colour fillColour, gfx::Colour outlineColour, uint8_t* linkedVariable)
+	: Slider(renderer, rect, name, fillColour, outlineColour, 0, 0, 255)
+	, linkedVariable_(linkedVariable)
+{
+	setForegroundColour(fillColour);
```

Comment:
What if linkedVariable is null?
**MUST**

Reply:
colour-linked textbox should not be defined unless it's linked explicitly to a colour value, so if nullptr should be invalid. Added an assert to ensure not nullptr.